### ISSUE TITLE
String interpolate environment variables

### ISSUE THREAD

**Issue by robinvdvleuten** [HenrikJoreteg/getconfig#28]
I use this module a lot and I ran into cases several times where I would like to interpolate any environment variables within a string. To make it make it more clearer, a small example;

```json
{
  "url": "http://${HOSTNAME}.com"
}
```

This would be a great addition to an already great library 👍

**Comment by nlf** [comment]
that's something i've considered.

in the past i've worked around this by just exporting another variable, i.e.
`export FULL_HOST="http://${HOSTNAME}.com"`
and then refer to `$FULL_HOST` in my config

this is kind of a special case since it really only makes sense for strings, and it also really only makes sense for required values. adding this support wouldn't be difficult, but i'm curious if you can think of any use case that pokes holes in my thinking there (in only allowing this for required strings)

**Comment by nlf** [comment]
talked it out with a friend and i'm actually going to one up you here

1) i was wrong about just strings, numbers also make sense (think things like port numbers and building a fully qualified url)
2) in addition to env var interpolation i'm _also_ going to allow config keys to refer to other config keys, something like:

```
{
  "port": "$PORT::int",
  "hostname": "${HOSTNAME}.com",
  "url": "http://${self.hostname}:${self.port}"
}
```

**Comment by nlf** [comment]
both of the above are now available in 4.1.0

**Comment by robinvdvleuten** [comment]
@nlf awesome! Thanks for the quick response 😄
